File: CS-Examples/04_RowsColumns/SetDefaultRowAndColumnStyle.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace SetDefaultRowAndColumnStyle
{
    public partial class Form1 : Form
    {

        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Create a cell style and set the color to yellow
            CellStyle style = workbook.Styles.Add("Mystyle");
            style.Color = Color.Yellow;

            // Set the default style for the first row using the created style
            sheet.SetDefaultRowStyle(1, style);

            // Set the default style for the first column using the created style
            sheet.SetDefaultColumnStyle(1, style);

            // Specify the output file name
            string result = "result.xlsx";

            // Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer(result);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }


}

File: CS-Examples/04_RowsColumns/SetDefaultRowAndColumnStyle.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace SetDefaultRowAndColumnStyle
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Create a new Workbook object.
            Dim workbook As New Workbook()

            'Get the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Create a cell style with "Mystyle" name.
            Dim style As CellStyle = workbook.Styles.Add("Mystyle")
            ' Set its color to Yellow.
            style.Color = Color.Yellow

            'Apply the above style to the default style of the first row.
            sheet.SetDefaultRowStyle(1, style)
            'Apply the above style to the default style of the first column.
            sheet.SetDefaultColumnStyle(1, style)


            'Specify the name for the resulting file.
            Dim result As String = "result.xlsx"

            'Save the workbook to a file with the specified name and Excel version (in this case, Excel 2010).
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()
            ExcelDocViewer(result)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class


End Namespace
